File: propbank/src/main/java/org/sqlunet/propbank/style/Colors.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.propbank.style;

import android.content.Context;
import android.graphics.Color;

import org.sqlunet.propbank.R;

import androidx.annotation.NonNull;

/**
 * Color values
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class Colors
{
	static public int thetaBackColor = Color.TRANSPARENT;
	static public int thetaForeColor = Color.TRANSPARENT;

	static public int relationBackColor = Color.TRANSPARENT;
	static public int relationForeColor = Color.TRANSPARENT;

	static public void setColorsFromResources(@NonNull final Context context)
	{
		// do not reorder : dependent on resource array order

		int[] palette = context.getResources().getIntArray(R.array.palette_pb);
		int i = 0;
		thetaBackColor = palette[i++];
		thetaForeColor = palette[i++];

		relationBackColor = palette[i++];
		//noinspection UnusedAssignment
		relationForeColor = palette[i++];
	}
}

File: propbank/src/main/java/org/sqlunet/propbank/style/Colors.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.propbank.style

import android.content.Context
import android.graphics.Color
import org.sqlunet.propbank.R

/**
 * Color values
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object Colors {

    var vnRoleBackColor = Color.TRANSPARENT

    var vnRoleForeColor = Color.TRANSPARENT

    var relationBackColor = Color.TRANSPARENT

    var relationForeColor = Color.TRANSPARENT

    fun setColorsFromResources(context: Context) {
        // do not reorder : dependent on resource array order
        val palette = context.resources.getIntArray(R.array.palette_pb)
        var i = 0
        vnRoleBackColor = palette[i++]
        vnRoleForeColor = palette[i++]
        relationBackColor = palette[i++]
        relationForeColor = palette[i]
        // increment i if colors added
    }
}
